### PR TITLE
Implement extract AST simplifications

### PR DESCRIPTION
**Pull request by SweetVishnya**
Part of #930

### CODE REVIEW

**Comment by SweetVishnya** [comment]
@JonathanSalwan, could you please rerun the pipeline?

File: src/testers/unittests/test_ast_simplification.py [hunk 1]
```diff
@@ -524,3 +524,109 @@ def test_xor3(self):
...
+
+    """Testing AST simplification"""
+
+    def setUp(self):
+        self.ctx = TritonContext()
+        self.ctx.setArchitecture(ARCH.X86_64)
+        self.ast = self.ctx.getAstContext()
+        self.ctx.setMode(MODE.AST_OPTIMIZATIONS, True)
+
+    def proof(self, n):
```
Comment (JonathanSalwan) [inline src/testers/unittests/test_ast_simplification.py:599]: never used

Comment (SweetVishnya) [reply src/testers/unittests/test_ast_simplification.py:599]: Done

File: src/libtriton/ast/astContext.cpp [hunk 2]
```diff
@@ -777,11 +777,114 @@ namespace triton {
...
+             * Optimization: If we extract from the node being extended, just
+             * return the node
+             *
+             * ((_ extract 31 0) ((_ zero_extend 32) (_ bv1 32))) => (_ bv1 32)
+             *
+             * ((_ extract 7 0) ((_ sign_extend 32) (_ bv1 32))) =>
+             * ((_ extract 7 0) (_ bv1 32))
+             **/
+            n = n->getChildren()[1];
+            auto sz = n->getBitvectorSize();
```
Comment (SweetVishnya) [inline src/libtriton/ast/astContext.cpp:848]: Found an error, sz is 32 for 8-bit symbolic memory reference. Trying to fix it now.

Comment (SweetVishnya) [reply src/libtriton/ast/astContext.cpp:848]: There is actually no error.

**Pull request merged**